[R-package] CMake errors in MinGW builds on Windows

See [the most recent build on `master`](https://ci.appveyor.com/project/guolinke/lightgbm/builds/32728411/job/jpvhc1pkgghirltt). There are some errors being generated when building the R package using MinGW. This isn't causing any user-facing issues as far as I know, since it doesn't cause installation to fail, but we should fix these warnings so that they don't waste users' time or lead to unnecessary bug reports.

```text
installing via 'install.libs.R' to C:/projects/lightgbm/lightgbm.Rcheck/00LOCK-lightgbm/00new/lightgbm
[1] "Trying to build with MinGW"
CMake Error at C:/Program Files (x86)/CMake/share/cmake-3.16/Modules/CMakeMinGWFindMake.cmake:12 (message):
...
...
...
CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
```

[This StackOverflow post](https://stackoverflow.com/a/32922266/3986677) suggested that maybe putting MinGW on `PATH` can fix this but so far I haven't had luck with it.

--- a/docs/FAQ.rst
+++ b/docs/FAQ.rst
@@ -186,6 +186,18 @@ The appropriate splitting strategy depends on the task and domain of the data, i
 
 LightGBM supports loading data from zero-based LibSVM format file directly.
 
+14. Why cmake cannot find the compiler when compiling LightGBM with MinGW?
+--------------------------------------------------------------------------
+
+.. code-block:: bash
+
+    CMake Error: CMAKE_C_COMPILER not set, after EnableLanguage
+    CMake Error: CMAKE_CXX_COMPILER not set, after EnableLanguage
+
+This is a known issue of `cmake` when using MinGW. The easiest solution is to run again `cmake` to bypass the one time stopper from `cmake`.
+
+See `Microsoft/LightGBM#3060 <https://github.com/microsoft/LightGBM/issues/3060#issuecomment-62633853>`__ for more details.
+
 ------
 
 R-package

